python/datasources/acs_health_insurance.py

```diff
@@ -0,0 +1,612 @@
+        json_formatted_response = resp.json()
+        state_fips = {}
+        for row in json_formatted_response[1::]:
+            state_name, fip_code = row
+            state_fips[fip_code] = state_name
+
+        self.state_fips = state_fips
+
+    # Pull the County Fips map Code->Name from ACS
+    def get_county_fips_mapping(self):
+        params = {'for': 'county', "get": "NAME"}
+        resp = requests.get(self.base_url, params=params)
+        json_formatted_response = resp.json()
+        county_fips = {}
+        for row in json_formatted_response[1::]:
+            county_name, state_fip, county_fip = row
+
+            county_fips[(state_fip, county_fip)] = county_name
+
+        self.county_fips = county_fips
+
+    # Given an ACS formatted param string, query the server, log and return the response as JSON
+    # Note: the method is defined this way to be similar to the Upload_to_GCS_util method.
+    def get_acs_data_from_variables(self, params):
+        resp = requests.get(self.base_url, params=params)
+        return resp.json()
+
+    #   Get Health insurance data from either GCS or Directly, and aggregate the data in memory
+    def get_health_insurance_data_by_race(self, use_gcs=False, gcs_bucket=None):
+        if use_gcs:
+            for race in RACE:
+                # Get cached data from GCS
+                state_data = gcs_to_bq_util.load_values_as_json(
+                    gcs_bucket, self.get_filename(None, race, False))
+                county_data = gcs_to_bq_util.load_values_as_json(
+                    gcs_bucket, self.get_filename(None, race, True))
+
+                # Aggregate in Memory
+                self.accumulate_acs_data(state_data)
+                self.accumulate_acs_data(county_data)
+        else:
+            for prefix in HEALTH_INSURANCE_BY_RACE_GROUP_PREFIXES:
+                # Get ACS data from API
+                state_data = self.get_acs_data_from_variables(format_params(
+                    prefix, HEALTH_INSURANCE_BY_RACE_GROUP_SUFFIXES))
+                county_sex_data = self.get_acs_data_from_variables(format_params(
+                    prefix, HEALTH_INSURANCE_BY_RACE_GROUP_SUFFIXES, True))
+
+                # Aggregate in Memory
+                self.accumulate_acs_data(state_data)
+                self.accumulate_acs_data(county_sex_data)
+
+    # Get Health insurance By Sex from either API or GCS and aggregate it in memory
+    def get_health_insurance_data_by_sex(self, use_gcs=False, gcs_bucket=None):
+        if(use_gcs):  # LOAD JSON BLOBS FROM GCS
+            male_state_data = gcs_to_bq_util.load_values_as_json(
+                gcs_bucket, self.get_filename(Sex.MALE, None, False))
+            female_state_data = gcs_to_bq_util.load_values_as_json(
+                gcs_bucket, self.get_filename(Sex.FEMALE, None, False))
+            male_county_data = gcs_to_bq_util.load_values_as_json(
+                gcs_bucket, self.get_filename(Sex.MALE, None, True))
```

Comment:
It would be great if you could add the name for arguments which are "hanging" e.g. change self.get_filename(Sex.Male, None, False) to self.get_filename(Sex.Male, race=None, is_county=False). It would be amazing if you could do this throughout the file wherever you have these hanging arguments, it helps quite a bit with readability / being able to follow the flow of the logic imo